file.c

```diff
@@ -5363,61 +5363,183 @@ test_check(int n, int argc, VALUE *argv)
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'g'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> has the \CF{setgid} bit set (false in Windows).</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'G'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and has a group ownership equal to the caller's group.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'k'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and has the sticky bit set.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'l'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is a symbolic link.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'M'</tt></th>
+ *      <th><tt>Time</tt></th>
+ *      <td>Last modification time for <tt>file1</tt>.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'o'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is owned by the caller's effective uid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'O'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is owned by the caller's real uid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'p'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is a fifo.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'r'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> is readable by the effective uid/gid of the caller.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'R'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether file is readable by the real uid/gid of the caller.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'s'</tt></th>
+ *      <th><tt>Integer</tt></br>or <tt>nil</tt></th>
+ *      <td>Size of <tt>file1</tt> if non-empty, otherwise <tt>nil</tt>.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'S'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is a socket.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'u'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> has the setuid bit set.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'w'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is writable by the effective uid/gid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'W'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is writable by the real uid/gid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'x'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is executable by the effective uid/gid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'X'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and is executable by the real uid/gid.</td>
+ *    </tr>
+ *    <tr>
+ *      <th><tt>'z'</tt></th>
+ *      <th>Boolean</th>
+ *      <td>Whether <tt>file1</tt> exists and has a zero length.</td>
+ *    </tr>
+ *  </table>
```

Comment:
I'd suggest not to use tables.

```suggestion
 *  <tt>"A"</tt> (Time)    :: Last access time for file1
 *  <tt>"b"</tt> (boolean) :: True if file1 is a block device
 *  <tt>"c"</tt> (boolean) :: True if file1 is a character device
 *  <tt>"C"</tt> (Time)    :: Last change time for file1
 *  <tt>"d"</tt> (boolean) :: True if file1 exists and is a directory
 *  <tt>"e"</tt> (boolean) :: True if file1 exists
 *  <tt>"f"</tt> (boolean) :: True if file1 exists and is a regular file
 *  <tt>"g"</tt> (boolean) :: True if file1 has the setgid bit set
 *                            (false under NT)
 *  <tt>"G"</tt> (boolean) :: True if file1 exists and has a group
 *                            ownership equal to the caller's group
 *  <tt>"k"</tt> (boolean) :: True if file1 exists and has the sticky
 *                            bit set
 *  <tt>"l"</tt> (boolean) :: True if file1 exists and is a symbolic link
 *  <tt>"M"</tt> (Time)    :: Last modification time for file1
 *  <tt>"o"</tt> (boolean) :: True if file1 exists and is owned by the
 *                            caller's effective uid
 *  <tt>"O"</tt> (boolean) :: True if file1 exists and is owned by the
 *                            caller's real uid
 *  <tt>"p"</tt> (boolean) :: True if file1 exists and is a fifo
 *  <tt>"r"</tt> (boolean) :: True if file1 is readable by the effective
 *                            uid/gid of the caller
 *  <tt>"R"</tt> (boolean) :: True if file is readable by the real uid/gid
 *                            of the caller
 *  <tt>"s"</tt> (Integer / +nil+):: If file1 has nonzero size, return the
 *                                   size, otherwise return nil
 *  <tt>"S"</tt> (boolean) :: True if file1 exists and is a socket
 *  <tt>"u"</tt> (boolean) :: True if file1 has the setuid bit set
 *  <tt>"w"</tt> (boolean) :: True if file1 exists and is writable by the
 *                            effective uid/gid
 *  <tt>"W"</tt> (boolean) :: True if file1 exists and is writable by the
 *                            real uid/gid
 *  <tt>"x"</tt> (boolean) :: True if file1 exists and is executable by the
 *                            effective uid/gid
 *  <tt>"X"</tt> (boolean) :: True if file1 exists and is executable by the
 *                            real uid/gid
 *  <tt>"z"</tt> (boolean) :: True if file1 exists and has a zero length
```

Reply:
I have learned to avoid labeled lists, which are sometimes rendered in unexpected ways (e.g., continuation lines not properly indented).

Reply:
There really are three things here, not two, so I think tables are appropriate;  things line up better.

Reply:
IMO the table is much easier to read than this kind of list.

Reply:
Using the `<table>` and related tags here also makes the `ri` output very unappealing.

<img width="542" alt="image" src="https://user-images.githubusercontent.com/17083/193616724-dccdc54d-ea1f-4de3-b74a-40430b47f318.png">

Reply:
Ah, excellent point.  So a '|'-based text table will look ok in both ri and html.  What about a labeled list in ri?  Should we be avoiding that?

Reply:
The label list looks good:

<img width="546" alt="image" src="https://user-images.githubusercontent.com/17083/193622418-0104e89f-4997-47ce-bdfa-ab9976804609.png">

@nobu mentions GFM support in another comment, so presumably, the GFM formatted table would look OK too, albeit annoying to indent and update.

```
| Character       | Return type      | Meaning                          |
|-----------------|------------------|----------------------------------|
| C               | Time             | Last modification time for file1 |
```

Screenshot of this comment with raw GFM formatted table:
<img width="771" alt="image" src="https://user-images.githubusercontent.com/17083/193623438-2a2b7cb6-6eef-4066-b686-158c911a0b5b.png" border="1">

Reply:
The GFM tables have no way to merge multiple lines to one cell (or continue a cell to the next line)
In the following example, `"g"` and `"G"` rows are split as different rows.

| Cmd | Returns | Meaning                                    |
|-----|---------|--------------------------------------------|
| "A" | Time    | Last access time for file1                 |
| "b" | boolean | True if file1 is a block device            |
| "c" | boolean | True if file1 is a character device        |
| "C" | Time    | Last change time for file1                 |
| "d" | boolean | True if file1 exists and is a directory    |
| "e" | boolean | True if file1 exists                       |
| "f" | boolean | True if file1 exists and is a regular file |
| "g" | boolean | True if file1 has the setgid bit           |
|     |         | set (false under NT)                       |
| "G" | boolean | True if file1 exists and has a group       |
|     |         | ownership equal to the caller's group      |

We have to write very long lines to avoid this.

Reply:
I have opened https://github.com/ruby/ruby/pull/6485._